packages/web/src/components/createCell.tsx

```diff
@@ -345,13 +350,23 @@ export function createCell<
 
     if (error) {
       if (Failure) {
+        // errorCode is not part of the type returned by useQuery
+        // but it is returned as part of the queryResult
+        type QueryResultWithErrorCode = typeof queryResult & {
+          errorCode: string
+        }
+
         return (
           <Failure
             error={error}
-            errorCode={error.graphQLErrors?.[0]?.extensions?.['code'] as string}
+            errorCode={
+              // Use the ad-hoc QueryResultWithErrorCode type to access the errorCode
+              (queryResult as QueryResultWithErrorCode).errorCode ??
```

Comment:
Not sure that adding an ad-hoc type is the best way to handle the `errorCode`. would love feedback on this fix 🤓

Reply:
This is a change to existing behavior, right? Where exactly is `errorCode` coming from? Why weren't we using it before?

Reply:
This showed up when running the test suite on this test: https://github.com/redwoodjs/redwood/blob/main/packages/web/src/components/createCell.test.tsx#L260

Before my change the `queryRest` destructuring was overriding the `errorCode` prop:

```tsx
          <Failure
            error={error}
            errorCode={error.graphQLErrors?.[0]?.extensions?.['code'] as string}
            {...props}
            updating={loading}
            {...queryRest} // <---- Overwrites errorCode because it's one of `queryRest` members. it also overwrites error
          />
```

But because it's now a part of `queryResult` the `errorCode` is never set. So the test referenced above fails.

Should we be destructuring and overwriting `error` and `errorCode`?